compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -360,6 +360,20 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    crate fn hir_id_sole_block_element(
+        &self,
+        hir_id: hir::HirId,
+    ) -> Option<&'tcx rustc_hir::Expr<'tcx>> {
+        let node: Option<Node<'_>> = self.tcx.hir().find(hir_id);
+        match node {
+            Some(Node::Expr(rustc_hir::Expr {
+                kind: rustc_hir::ExprKind::Block(block, ..),
```

Comment:
Nit: `rust_hir` -> `hir`
```suggestion
    ) -> Option<&'tcx hir::Expr<'tcx>> {
        let node: Option<Node<'_>> = self.tcx.hir().find(hir_id);
        match node {
            Some(Node::Expr(hir::Expr {
                kind: hir::ExprKind::Block(block, ..),
```